app/policies/bookmark_policy.rb

```diff
@@ -0,0 +1,10 @@
+class BookmarkPolicy < ApplicationPolicy
+  # ApplicationPolicy defaults will work for bookmarks. Nothing needed other than class definition
+  def update?
+    user.present? && (record.topic.user == user)
+  end
+  
+  def destroy?
+    user.present? && (record.topic.user == user)
```

Comment:
@ianroberts131 Consider adding a `has_one :user, through: :topic` association to `Bookmark`. Benefits of this are that you'll be able to call `bookmark.user` rather than `bookmark.topic.user`, meaning the custom `update?` and `destroy?` methods can be removed as the ApplicationPolicy implementations will work.